astlib/keyword.ml

```diff
@@ -57,3 +57,43 @@ let is_keyword = function
   | "lsr" -> true
   | "asr" -> true
   | _ -> false
+
+let apply_keyword_edition ~cli () =
+  let from_ocaml_param =
+    match Sys.getenv "OCAMLPARAM" with
+    | s -> (
+        let items =
+          if String.equal s "" then []
+          else
+            (* cf. Compenv.parse_args *)
```

Comment:
The problem is we can't make a dep on compilerlibs?

Reply:
I think the exposed version of this is a bit too high level and is used to parse the CLI args of a driver so it wouldn't work well for us. Maybe @dra27 can confirm?

Reply:
Right, it might be good to ask upstream to expose some of the internals if downstream libraries like ppxlib have to depend on that behaviour. Having just seen the issues caused in #540 by _copying_ pieces of the compiler's internals into ppxlib, I think a long-term strategy of forcing the compiler to expose these things and help with long term stability would be nice?

Not a blocker for this PR because it seems the pieces we need are not necessarily exposed (but maybe should be? for the 5.3 release?).

Reply:
Yeah that's a good point! We should then at least move the bits we'd like exposed in the compiler to `Astlib`.

@dra27 do you think we could reasonably expect something like `read_OCAMLPARAM`, returning a pair of args bindings lists, to be exposed at some point in the future?

Reply:
Or maybe even `apply_keyword_edition` directly!

Reply:
CC @dra27 any thoughts on that?

Reply:
It's definitely too late to be making compiler-libs changes for 5.3, but that obviously preclude doing that for 5.4.

I agree with the general concern at the duplication (I was concerned about it when hacking up the first version, too). However, it's worth remembering that it's _extremely_ hard for OCaml itself to change the interpretation of the variable, so the details are very unlikely to change - i.e. we're duplicating a little bit of code which deals with a very non-internal detail of the compiler here.

Reply:
Yeah that's a good point!

Since we can't reasonably get this into 5.3, I'd say this PR is good to go! What do you think @patricoferris ?

Reply:
Yep! Good to go :))